src/multi-selector.js

```diff
@@ -77,6 +78,15 @@ const MultiSelector = React.createClass({
 		})
 	},
 
+	shouldComponentUpdate: function(nextProps, nextState) {
+		return nextState.searchValue !== this.state.searchValue
+			|| nextProps.initialSelectedItems !== this.state.selectedItems
+			|| nextProps.items !== this.props.items
+			|| nextState.dialogDisplayed !== this.state.dialogDisplayed
+			|| nextState.activeIndex !== this.state.activeIndex
+			|| nextState.mouseIndex !== this.state.mouseIndex;
```

Comment:
I was kinda under the impression that if I did that there would be no difference from the default way React determines whether to update. So I tried to only check the values we really care about? It's highly possible i'm going about this the wrong way entirely